crates/walrus-sdk/src/config/committees_refresh_config.rs

```diff
@@ -45,40 +46,36 @@ impl CommitteesRefreshConfig {
     /// returns the [`CommitteesRefresherHandle`].
     pub async fn build_refresher_and_run(
         &self,
-        sui_client: impl ReadClient + 'static,
+        sui_client: impl ReadClient + Clone + 'static,
     ) -> Result<CommitteesRefresherHandle> {
-        let (mut refresher, handle) = build_refresher_and_handle(sui_client, self.clone()).await?;
+        let n_shards = sui_client
+            .clone()
+            .n_shards()
+            .await
+            .context("failed to determine n_shards before starting refresher")?;
+
+        let notify = Arc::new(Notify::new());
+        let (req_tx, req_rx) = mpsc::channel(self.refresher_channel_size);
+        let handle = CommitteesRefresherHandle::new(notify.clone(), req_tx, n_shards);
+        let config = self.clone();
```

Comment:
Here, is the main change. We no longer await the `build_refresher_and_handle()` anymore, but instead create and run refresher in the background task.